test/createvolume_tester.py

```diff
@@ -1,122 +1,146 @@
-
-    def get_request_params(self):
-        return {"Name": "test-vol-001",
-                "Opts": {}}
-
-    def setup_mock_objects(self):
-        mock_etcd = self.mock_objects['mock_etcd']
-        mock_etcd.get_vol_byname.return_value = 'Imran'
-
-        # try:
-        #     self._set_flash_cache_policy_in_vvs(flash_cache, vvs_name)
-        #     self.client.addVolumeToVolumeSet(vvs_name, volume_name)
-        # except Exception as ex:
-        #     # Cleanup the volume set if unable to create the qos rule
-        #     # or flash cache policy or add the volume to the volume set
-        # TODO: Ensure that deleteVolumeSet is invoked here
-        #     self.client.deleteVolumeSet(vvs_name)
-        # TODO: Handle this
-        #     raise exception.PluginException(ex)
-
-
-# More cases of flash cache
-# 1.
-# if flash_cache:
-#     try:
-#         self.client.modifyVolumeSet(vvs_name,
-#                                     flashCachePolicy=flash_cache)
-#         LOG.info(_LI("Flash Cache policy set to %s"), flash_cache)
-#     except Exception as ex:
-#         LOG.error(_LE("Error setting Flash Cache policy "
-#                       "to %s - exception"), flash_cache)
-#         exception.PluginException(ex)
+# import mock
+import testtools
+import fake_3par_data as data
+
+import hpe_docker_unit_test as hpedockerunittest
+
+
+class CreateVolumeUnitTest(hpedockerunittest.HpeDockerUnitTestExecutor):
+
+    # This function carries out common steps needed by create-volume for
+    # different mock-etcd configurations, docker configuration, create-volume
+    # requests and checking of responses for success/failure
+    def run_test(self, test_case):
+        self._test_case = test_case
+        # This is important to set as it is used by the base class to
+        # take decision which driver to instantiate
+        self._protocol = test_case.protocol
+        operation = 'volumedriver_create'
+        self._test_operation(operation)
+
+    def setup_mock_objects(self):
+        mock_etcd = self.mock_objects['mock_etcd']
+        mock_etcd.get_vol_byname.return_value = None
+
+    def override_configuration(self, config):
+        pass
+
+
+# class TestCreateVolumeDefault(CreateVolumeUnitTest, testtools.TestCase):
+class TestCreateVolumeDefault(CreateVolumeUnitTest):
+    def check_response(self, resp):
+        self._test_case.assertEqual(resp, {u"Err": ''})
+
+    def get_request_params(self):
+        return {"Name": "test-vol-001",
+                "Opts": {}}
+
+    def setup_mock_objects(self):
+        mock_etcd = self.mock_objects['mock_etcd']
+        mock_etcd.get_vol_byname.return_value = None
+
+
+# Provisioning = Full
+class TestCreateThickVolume(CreateVolumeUnitTest):
+    def check_response(self, resp):
+        self._test_case.assertEqual(resp, {u"Err": ''})
+
+    def get_request_params(self):
+        return {"Name": "test-vol-001",
+                "Opts": {'provisioning': data.FULL}}
```

Comment:
size is missing

Reply:
Intentionally size has been dropped for it to use default size. Can add it though.